src/masternode.cpp

```diff
@@ -2,108 +2,112 @@
 // Distributed under the MIT/X11 software license, see the accompanying
 // file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
-#include "consensus/validation.h"
 #include "activemasternode.h"
+#include "consensus/validation.h"
 #include "darksend.h"
 #include "init.h"
+#include "governance.h"
 #include "masternode.h"
 #include "masternode-payments.h"
 #include "masternode-sync.h"
 #include "masternodeman.h"
-#include "governance.h"
 #include "util.h"
-#include "sync.h"
-#include "addrman.h"
-#include <boost/lexical_cast.hpp>
 
-// keep track of the scanning errors I've seen
-map<uint256, int> mapSeenMasternodeScanningErrors;
+#include <boost/lexical_cast.hpp>
 
-CMasternode::CMasternode()
-{
-    LOCK(cs);
-    vin = CTxIn();
-    addr = CService();
-    pubkey = CPubKey();
-    pubkey2 = CPubKey();
-    vchSig = std::vector<unsigned char>();
-    activeState = MASTERNODE_ENABLED;
-    sigTime = GetAdjustedTime();
-    lastPing = CMasternodePing();
-    nTimeLastPaid = 0;
-    nBlockLastPaid = 0;
-    nCacheCollateralBlock = 0;
-    unitTest = false;
-    allowFreeTx = true;
-    protocolVersion = PROTOCOL_VERSION;
-    nLastDsq = 0;
-    nScanningErrorCount = 0;
-    nLastScanningErrorBlockHeight = 0;
-    lastTimeChecked = 0;
-}
 
-CMasternode::CMasternode(const CMasternode& other)
-{
-    LOCK(cs);
-    vin = other.vin;
-    addr = other.addr;
-    pubkey = other.pubkey;
-    pubkey2 = other.pubkey2;
-    vchSig = other.vchSig;
-    activeState = other.activeState;
-    sigTime = other.sigTime;
-    lastPing = other.lastPing;
-    nTimeLastPaid = other.nTimeLastPaid;
-    nBlockLastPaid = other.nBlockLastPaid;
-    nCacheCollateralBlock = other.nCacheCollateralBlock;
-    unitTest = other.unitTest;
-    allowFreeTx = other.allowFreeTx;
-    protocolVersion = other.protocolVersion;
-    nLastDsq = other.nLastDsq;
-    nScanningErrorCount = other.nScanningErrorCount;
-    nLastScanningErrorBlockHeight = other.nLastScanningErrorBlockHeight;
-    lastTimeChecked = 0;
-}
-
-CMasternode::CMasternode(const CMasternodeBroadcast& mnb)
-{
-    LOCK(cs);
-    vin = mnb.vin;
-    addr = mnb.addr;
-    pubkey = mnb.pubkey;
-    pubkey2 = mnb.pubkey2;
-    vchSig = mnb.vchSig;
-    activeState = MASTERNODE_ENABLED;
-    sigTime = mnb.sigTime;
-    lastPing = mnb.lastPing;
-    nTimeLastPaid = 0;
-    nBlockLastPaid = 0;
-    nCacheCollateralBlock = 0;
-    unitTest = false;
-    allowFreeTx = true;
-    protocolVersion = mnb.protocolVersion;
-    nLastDsq = mnb.nLastDsq;
-    nScanningErrorCount = 0;
-    nLastScanningErrorBlockHeight = 0;
-    lastTimeChecked = 0;
-}
+CMasternode::CMasternode() :
+    vin(CTxIn()),
+    addr(CService()),
+    pubKeyCollateralAddress(CPubKey()),
+    pubKeyMasternode(CPubKey()),
+    lastPing(CMasternodePing()),
+    vchSig(std::vector<unsigned char>()),
```

Comment:
Yes, this is mostly for the sake of keeping everything in the same order, rather then actual initialization.